### PR TITLE
Improve update database

### PR DESCRIPTION
**Pull request by lbcd**
Improve Update database command to allow update from manual link provided by MaxMind

Case when you have bought a license.
I can only test for country. I have no license for city and others.

### CODE REVIEW

**Comment by cravler** [comment]
please add archive structure

**Comment by lbcd** [comment]
Do you have an exemple?

**Comment by cravler** [comment]
no

**Comment by lbcd** [comment]
That will no help us to try to collaborate to make a better Geoip bundle...
I will try to find by myself.

**Comment by cravler** [comment]
I need original structure of the archive, to know how decompress should work
Can you make some screenshot?

**Comment by lbcd** [comment]
Oh sorry I didn't understand. I send it to you by email.

File: Command/UpdateDatabaseCommand.php [hunk 1]
```diff
@@ -112,16 +132,89 @@ private function downloadFile($source)
...
+                }
+            }
+
+            $tarFile = new \PharData(str_replace ('.gz', '', $fileName));
+            $tarFile->extractTo($this->geoIpTmpExtractedFilesDirectory, null, true);
+
+            // Get directory created to finally have the mmdb file to copy
+            if (is_dir($this->geoIpTmpExtractedFilesDirectory)) {   // /tmp/MaxMind/ExtractedFiles
+                if ($dh = opendir($this->geoIpTmpExtractedFilesDirectory)) {
+                    while (($file = readdir($dh)) !== false) {
```
Comment (lbcd) [inline Command/UpdateDatabaseCommand.php:195]: Loop is not needed here. I should remove it. But it works.

File: Command/UpdateDatabaseCommand.php [hunk 2]
```diff
@@ -112,16 +132,89 @@ private function downloadFile($source)
...
+                }
+                unlink($filename);
+            }
+
+            $p = new \PharData($fileName);
+            $p->decompress(); // creates the .tar file by ungziping it
+
+            // Delete old uncompressed directory  -- no recursive function to prevent bad deletion. We must have 2 directory level
+            if (is_dir($this->geoIpTmpExtractedFilesDirectory)) {   // /tmp/MaxMind/ExtractedFiles
+                if ($dh = opendir($this->geoIpTmpExtractedFilesDirectory)) {
```
Comment (lbcd) [inline Command/UpdateDatabaseCommand.php:156]: Can be simplified in one line "if".

**Comment by cravler** [comment]
On the basis of your `pull request` I have made changes to the Bundle code.

**Comment by lbcd** [comment]
Ok thank you. I will try that.

**Comment by lbcd** [comment]
It works. Thank you. Can you create a release like 1.1.1 so I can set it in my composer.json ?

**Comment by cravler** [comment]
Done

**Comment by lbcd** [comment]
Thank you very much for you reactivity.

**Pull request closed without merging**